### PR TITLE
FORMS-8202 Allow Null dataRef in exported JSON

### PR DESCRIPTION
**Pull request by barshat7**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java [hunk 1]
```diff
@@ -404,4 +419,17 @@ public FormComponentData getData() {
         return componentData;
     }
 
+    public static class CustomSerializer extends JsonSerializer<String> {
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java:422]: Please make it private. Rename the `CustomSerializer` to `DataRefSerializer`

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -128,6 +128,20 @@
                                                     fieldDescription="You can specify a Document of Record Bind Reference to link an adaptive form field with a document of record field. When you fill any value in the linked field of an adaptive form, the value is also displayed in the linked field of the corresponding document of a record."
                                                     fieldLabel="Document of Record bind reference"
                                                     name="./dorBindRef"/>
+                                            <unboundFormElement
+                                              jcr:primaryType="nt:unstructured"
+                                              sling:resourceType="granite/ui/components/coral/foundation/form/checkbox"
+                                              granite:class="cmp-adaptiveform-base__unboundformelement"
+                                              name="./unboundFormElement"
+                                              text="Mark as Unbound Form Element"
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/_cq_dialog/.content.xml:136]: Can you add some description or help text to this ? Please sync with khushwant on this

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java [hunk 3]
```diff
@@ -404,4 +419,17 @@ public FormComponentData getData() {
...
+
+        @Override
+        public void serialize(String s, JsonGenerator jsonGenerator,
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java:425]: Please add a junit test case for this

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/_cq_dialog/.content.xml (hunk 2 above)
Comment (barshat7) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/_cq_dialog/.content.xml:136]: Will raise a separate ticket and discuss with khushwant

**Pull request merged**